### PR TITLE
pkg/server: export sql.new_conns in /_status/load

### PR DESCRIPTION
**Pull request by jaylim-crl**
The load status endpoint is currently used by the autoscaler in CockroachDB
Cloud for our Serverless offering. Previously, the metrics that we were relying
on were insufficient as they causes state oscillations in SQL pods where
connections have not gotten past the authentication phase. This commit updates
the load status endpoint to also export sql.new_conns as a counter, which we
will rely on in the autoscaler once this has been rolled out.

Part of https://cockroachlabs.atlassian.net/browse/CC-28796.

Epic: none

Release note: None

### CODE REVIEW

File: pkg/server/load_endpoint.go [hunk 1]
```diff
@@ -50,6 +50,17 @@ type loadEndpoint struct {
 	mainMetricSource metricMarshaler
 }
 
+// customLoadEndpointMetricsSet returns the set of all custom metric names that
+// the load endpoint should export.
+func customLoadEndpointMetricsSet() map[string]struct{} {
```
Comment (JeffSwenson) [inline pkg/server/load_endpoint.go:55]: nit: the `custom` prefix doesn't really make sense to me. It makes me think of the custom metric page in dbconsole. If anything, in the context of the load endpoint, the custom metrics would be the metrics that are not exported as part of the /status/_vars endpoint. What do you think of changing the name to `loadEndpointMetrics`?

**Review by jeffswenson — APPROVED** [review]
LGTM

**Review by jaylim-crl — COMMENTED** [review]
TFTR! 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/125697#-:-O-NAwcx51pxwSRnxu9a:bz5m8k4)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @andy-kimball and @JeffSwenson)

___
*[`pkg/server/load_endpoint.go` line 55 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/125697#gh-1640219962:-O-NAxTZ5xxYVnJBiPcZ:bk6jccq) ([raw file](https://github.com/cockroachdb/cockroach/blob/08b3355b8d7f09cb409f1f03f249d88aba2ab599/pkg/server/load_endpoint.go#L55)):*
<details><summary><i>Previously, JeffSwenson (Jeff Swenson) wrote…</i></summary><blockquote>

nit: the `custom` prefix doesn't really make sense to me. It makes me think of the custom metric page in dbconsole. If anything, in the context of the load endpoint, the custom metrics would be the metrics that are not exported as part of the /status/_vars endpoint. What do you think of changing the name to `loadEndpointMetrics`?
</blockquote></details>

I updated this to `additionalLoadEndpointMetricsSet`. I think `additional` here makes sense since the load endpoint was originally used for CPU-related metrics.

**Comment by jaylim-crl** [comment]
bors r=JeffSwenson

**Pull request merged**